src/cascadia/TerminalControl/TermControl.idl

```diff
@@ -12,6 +12,12 @@ import "ControlCore.idl";
 namespace Microsoft.Terminal.Control
 {
 
+    enum CursorDisplayState
+    {
+        Default,
+        Shown
+    };

```

Comment:
Will there be more states or will this remain a boolean? In case of the latter, I think some code can be expressed more elegantly with a boolean like
```cpp
control.CursorVisibility(
    activated
    ? Microsoft::Terminal::Control::CursorDisplayState::Shown
    : Microsoft::Terminal::Control::CursorDisplayState::Default
);
```
vs.
```cpp
control.IsCursorVisible(activated);
```

Reply:
(a half real suggestion that is somewhat relevant) you could also use the [Visibility](https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.visibility?view=winrt-22621) from XAML, if you want.

Reply:
Yea, `Default` is "the control gets to say if the cursor is visible or not", while `Shown` is "Please always show your cursor". 

When we "specced" this in Teams chat, we also left in a `Hidden` state, for a consumer who wanted to always hide the cursor. We obv didn't need that here.

I've kinda weirdly gotten attached to binary enums as bool values, just cause then the arg literally says what the value means, rather than having to look up what `true`/`false` means.

Reply:
But would it make sense to still rename it a bit to make it more obvious what it means? We have more cursor visibility states than this one after all (like blinking visibility, IME visibility, VT visibility, etc.) and the naming clashes with those IMO.
I personally think that `HideCursorOnBlur` improves on this as it ties the meaning of true/false to the function name. For an enum I think that `Always` instead of `Shown` and `HideOnBlur` instead of `Default` would be easier to understand (different names are probably even better).

I'll approve it in the meantime, but maybe this is something to still noodle about. ^^